Harvest/ui tweaks

y-lohse opened the pull request:
- Introduced a variant of the modal header where the layout is vertical instead of horizontal
- Corrected margins on one of the forms

Crash-- commented:
> Introduced a variant of the modal header where the layout is vertical instead of horizontal
 
Can you share a use case? screenshot ?

ptbrowne commented on packages/cozy-harvest-lib/src/components/KonnectorModalHeader.jsx:
```diff
@@ -2,16 +2,28 @@ import React from 'react'
+        <div
+          className={cx('u-flex-grow-1', vertical ? 'u-mb-half' : 'u-mr-half')}
+        >
+          <h3 className="u-title-h3 u-m-0">{title}</h3>
```
IMHO having to introduce a new API (`vertical`) and having to conditionally put utility classes shows that we are not working at the right level here.

What we seem to want here seems to switch between a stack based layout and a row based layout, with margins between items :

```
const Wrapper = this.props.wrapper // (Stack || Row)
<Wrapper spacing='m'>
</Wrapper>
```

ptbrowne replied:
What happened to this ?

y-lohse replied:
Wtf, I thought I posted a comment explaining the latest changes. Must have forgotten to send it..? Adding it again.

y-lohse commented:
Here's a screenshot of the other style:

<img width="585" alt="Capture d'écran 2019-10-17 11 16 54" src="https://user-images.githubusercontent.com/2261445/67012279-6859a200-f0f1-11e9-8e8f-84914a4e48a4.png">

I also agree in hindsight that the `vertical` API change wasn't the best. After talking with @joel-costa , it turns out that in the 2 cases I'm changing, the konnector icon and title shouldn't be a header at all, they are part of the modal's body, should scroll with it if it overflows, etc. So in the end I reverted the changes to `KonnectorModalHeader` and added the new content to the modals that needed the change.
